core/src/test/java/org/apache/iceberg/TestBaseIncrementalChangelogScan.java

```diff
@@ -0,0 +1,310 @@
+    Assert.assertEquals("Ordinal must match", 0, deletedDataFileTask.changeOrdinal());
+    Assert.assertEquals(
+        "Snapshot must match", secondSnapshot.snapshotId(), deletedDataFileTask.commitSnapshotId());
+    Assert.assertEquals("Data file must match", FILE_A.path(), deletedDataFileTask.file().path());
+    Assert.assertTrue("Must be no deletes", deletedDataFileTask.existingDeletes().isEmpty());
+  }
+
+  @Test
+  public void testExistingEntriesInNewDataManifestsAreIgnored() {
+    table
+        .updateProperties()
+        .set(MANIFEST_MIN_MERGE_COUNT, "1")
+        .set(MANIFEST_MERGE_ENABLED, "true")
+        .commit();
+
+    table.newAppend().appendFile(FILE_A).commit();
+
+    table.newAppend().appendFile(FILE_B).commit();
+
+    table.newAppend().appendFile(FILE_C).commit();
+
+    Snapshot thirdSnapshot = table.currentSnapshot();
+
+    ManifestFile manifest = Iterables.getOnlyElement(thirdSnapshot.dataManifests(table.io()));
+    Assert.assertTrue("Manifest must have existing files", manifest.hasExistingFiles());
+
+    IncrementalChangelogScan scan =
+        newScan()
+            .fromSnapshotInclusive(thirdSnapshot.snapshotId())
+            .toSnapshot(thirdSnapshot.snapshotId());
+
+    List<ChangelogScanTask> tasks = plan(scan);
+
+    Assert.assertEquals("Must have 1 task", 1, tasks.size());
+
+    AddedRowsScanTask task = (AddedRowsScanTask) Iterables.getOnlyElement(tasks);
+    Assert.assertEquals("Ordinal must match", 0, task.changeOrdinal());
+    Assert.assertEquals("Snapshot must match", thirdSnapshot.snapshotId(), task.commitSnapshotId());
+    Assert.assertEquals("Data file must match", FILE_C.path(), task.file().path());
+    Assert.assertTrue("Must be no deletes", task.deletes().isEmpty());
+  }
+
+  @Test
+  public void testManifestRewritesAreIgnored() throws IOException {
+    table.newAppend().appendFile(FILE_A).commit();
+
+    Snapshot firstSnapshot = table.currentSnapshot();
+
+    table.newAppend().appendFile(FILE_B).commit();
+
+    Snapshot secondSnapshot = table.currentSnapshot();
+
+    ManifestFile newManifest =
+        writeManifest(
+            "manifest-file.avro",
+            manifestEntry(ManifestEntry.Status.EXISTING, firstSnapshot.snapshotId(), FILE_A),
+            manifestEntry(ManifestEntry.Status.EXISTING, secondSnapshot.snapshotId(), FILE_B));
+
+    RewriteManifests rewriteManifests = table.rewriteManifests();
+
+    for (ManifestFile manifest : secondSnapshot.dataManifests(table.io())) {
+      rewriteManifests.deleteManifest(manifest);
+    }
+
+    rewriteManifests.addManifest(newManifest);
+
+    rewriteManifests.commit();
+
+    table.newAppend().appendFile(FILE_C).commit();
+
+    Snapshot fourthSnapshot = table.currentSnapshot();
+
+    List<ChangelogScanTask> tasks = plan(newScan());
+
+    Assert.assertEquals("Must have 3 tasks", 3, tasks.size());
+
+    AddedRowsScanTask firstTask = (AddedRowsScanTask) tasks.get(0);
```

Comment:
I should probably reconsider names here. Maybe, `task1` would be better? Then lines below would fit on one line.